ARCH.md

```diff
@@ -0,0 +1,49 @@
+Rust-Lightning is broken into a number of high-level structures with various APIs to hook
+them together, as well as a number of APIs for you to provide external data which you,
+the user, have to provide.
```

Comment:
The wording here is awkward with using "provide" twice.

Reply:
Yeah, I'd say `s/as well as a number of APIs for you to provide external data which you, the user, have to provide/as well as APIs for you (the user) to provide external data`